src/lexer/Tokenizer.php

```diff
@@ -77,22 +77,44 @@ public function digit()
     $buffer = [];
     $is_double = false;
 
+
     // Hexadecimal or binary
-    if ((int) $this->peek === 0 && in_array(strtolower($this->preview()), ['x','b'], true) && ctype_xdigit((string) $this->preview(2))) {
-      $this->readChar(); // 0
-      $type = strtolower($this->readChar()); // x or b
+    if ($this->peek === '0' && (in_array(strtolower($this->preview()), ['x', 'b']))) {
+      $buffer[] = $this->readChar(); // 0
+
+      // b and x are identifiers if there is no valid value after then
+      if (($this->peek === 'b' && (int) $this->preview() >> 1)         // Fake bin
+        || ($this->peek === 'x' && !ctype_xdigit($this->preview()))) { // Fake hex
+
+        $value = '0';
+        $this->column += 1;
+        return new Token(Tag::T_INTEGER, $this->symbol_table->add($value));
+      }
+
+      $buffer[] = $b_or_x = strtolower($this->readChar()); // b or x
+
+      switch ($b_or_x) {
+        case 'b':
+
+          do {
+            $buffer[] = $this->readChar();
+          } while (ctype_digit($this->peek) && !((int) $this->peek >> 1));
```

Comment:
We can't. It is a state machine. I have no control of the entire buffer of characters, but I must walk through the "tape", as a [DFA](https://en.wikipedia.org/wiki/Deterministic_finite_automaton).